contents/architecture/RESTArchitecturalStyle.md

```diff
@@ -0,0 +1,76 @@
+
+The quote below from Roy Fielding outlines what the name entails:
+> The name “Representational State Transfer” is intended to evoke an image of how a well-designed Web application behaves: a network of web pages (a virtual state-machine), where the user progresses through the application by selecting links (state transitions), resulting in the next page (representing the next state of the application) being transferred to the user and rendered for their use.
+
+## Why REST exists
+REST specifies the key designs of the Web. The design of the Web affects greatly how efficient, applicable and modifiable the Web can be. Hence, before understanding the REST principles, we need to understand the requirements of the Web i.e. what the Web is expected to be. 
+
+The Web is expected to be a place for convenient storing and sharing of information by its users. Hence, to make the ideal Web, we need to consider the properties of its users and their information. The target audience of the Web are people in different organizations all over the world connected via the Internet. Their organizations would have different requirements for their information (e.g. the authentication for access of information), and the machines they use to access the Web might be very different, with different operating systems and requesting different file formats. These people’s information also has a large variety in terms of the content type and the format. Hence, the Web needs to have the following properties:
+ * the Web should have a simple, uniform interface to present various types of information (e.g. video, audio, graphics, text etc.) 
+ * the Web should be efficient in transmitting information 
+ * the Web should be able to evolve
+ * the Web should not have a central entity to control the whole system, which means there is no control over the amount and the content of information outside the boundary of the organizations using the Web. Hence, 1) the Web elements should be able to cope with unexpected load 2) the Web should be able to communicate authentication data and authorization controls to sieve out the malicious information
+ * the Web should allow its element to undergo incremental changes for evolution 
+
+REST is the abstraction of the design that satisfies the above requirements.
+
+## Details of REST
+
+REST is an architectural style. Fielding defined architectural style as “a coordinated set of architectural constraints that restricts the roles/features of architectural elements and the allowed relationships among those elements within any architecture that conforms to that style”. 
+
+After surveying some common architectures for network-based applications on how the architectural constraints induce their corresponding architectural properties, Fielding came up with REST. REST has 6 constraints which aim to induce the properties the Web should have. 
+
+### <a name="clientServer"></a>Client-Server
```

Comment:
Hmm...actually I think using [Markdown Anchors](https://gist.github.com/asabaylus/3071099) should work as well.